app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/address/AddressController.scala

```diff
@@ -53,7 +54,7 @@ trait AddressController[J <: JourneyStatus] {
 
   def validJourney(request: RequestWithSessionData[_]): Either[Result, (SessionData, J)]
 
-  def updateAddress(journey: J, address: Address): JourneyStatus
+  def updateAddress(journey: J, address: Address)(implicit hc: HeaderCarrier): EitherT[Future, Error, JourneyStatus]
 
```

Comment:
should return type be:`EitherT[Future, Error, J]` rather than `EitherT[Future, Error, JourneyStatus]`

Reply:
I don't think it is necessary to widen the type here

Reply:
I left it as `JourneyStatus` to allow for the cases where the update doesn't result in the same type of `JourneyStatus` as `J` - however, none of the implementing classes actually do that, and based on what happened with the `EmailController` having `JourneyStatus` might not work anyway. I'll change this to be `J` for now and worry about the case with a different journey type when it happens

Reply:
changed